### ISSUE TITLE
How to select "θ" for PGExplainer in detail

### ISSUE THREAD

**Issue by BillLiu96** [steveazzolin/gnn_logic_global_expl#2]
Hi,

I am trying to run GLGExplainer on other molecule graph datasets, but I am confused about how you select "_θ_". 

As far as I understand, the "_θ_" you use is to get local explanations by removing edges with lower weights for each original graph. 
From what I understand, I fed the local explanations generated using different "_θ_" values (in [min_weight, max_weight]) into the pre-trained GNN for classification. However, as "_θ_" gets smaller (i.e., fewer masked edges on the original graphs), the F1 score increases overall.

In your paper, it says "For Mutagenicity, over which PGExplainer was originally evaluated, we simply selected the threshold θ that maximises the F1 score of the local explainer over all graphs." Back on the configure file for Mutagenicity,  "manual_cut" is set to 0.37731856. 

If I understand correctly, for Mutagenicity, did you choose the smallest non-zero edge weight for _θ_? If not, looking forward to your specific instruction on how you select _θ_ for molecule datasets like Mutagenicity.

I would appreciate it if you could help.

**Comment by steveazzolin** [comment]
Hi,

In the original PGExplainer paper, the authors are computing the ROC_AUC of the explainer, given that they know the expected ground truth.
Specifically, they compute the ROC_AUC which takes into account multiple values of the threshold θ, giving you an idea of also how sensible the threshold is.

In our setting, instead, for different fixed values of θ we compute the F1 score of the Local Explainer (F1 and not Acc because it has less bias when we have class imbalance). Then, we kept the value of θ which maximizes the F1 score over the entire train set.

Hope this helps
